fission/supporttool/resources/kubernetes.go

```diff
@@ -0,0 +1,260 @@
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj corev1.Pod) {
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesHPA:
+		objs, err := res.client.AutoscalingV2beta1().HorizontalPodAutoscalers(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj autoscalingv2beta1.HorizontalPodAutoscaler) {
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesNode:
+		objs, err := res.client.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj corev1.Node) {
+				obj = nodeClean(obj)
+				// Node doesn't have namespace value, use name here
+				f := filepath.Clean(fmt.Sprintf("%v/%v", dumpDir, obj.Name))
+				getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	default:
+		log.Printf("Unknown type: %v", res.objType)
+	}
+}
+
+// serviceClean remove sensitive data(e.g. public IP, external name) from service objects
+func serviceClean(svc corev1.Service) corev1.Service {
+	svc.Spec.ExternalIPs = []string{}
+	svc.Spec.LoadBalancerIP = ""
+	svc.Spec.LoadBalancerSourceRanges = []string{}
+	svc.Spec.ExternalName = ""
+	return svc
+}
+
+func nodeClean(node corev1.Node) corev1.Node {
+
+	var nodeAddresses []corev1.NodeAddress
+	for _, address := range node.Status.Addresses {
+		// use whitelist to filter the necessary information for debugging
+		if address.Type == "InternalIP" || address.Type == "Hostname" {
+			nodeAddresses = append(nodeAddresses, address)
+		}
+	}
+	node.Status.Addresses = nodeAddresses
+
+	return node
+}
+
+type KubernetesPodLogDumper struct {
+	client        *kubernetes.Clientset
+	labelSelector string
+}
+
+func NewKubernetesPodLogDumper(clientset *kubernetes.Clientset, selector string) Resource {
+	return KubernetesPodLogDumper{
+		client:        clientset,
+		labelSelector: selector,
+	}
+}
+
+func (res KubernetesPodLogDumper) Dump(dumpDir string) {
+	l, err := res.client.CoreV1().
+		Pods(metav1.NamespaceAll).
+		List(metav1.ListOptions{LabelSelector: res.labelSelector})
+	if err != nil {
+		log.Printf("Error getting controller list: %v", err)
+		return
+	}
+
+	wg := &sync.WaitGroup{}
+
+	for _, p := range l.Items {
+		wg.Add(1)
+
+		go func(pod corev1.Pod) {
+			defer wg.Done()
+
+			if !fission.IsReadyPod(&pod) {
+				fmt.Printf("Pod %v is not in ready state, ignore it\n", pod.Name)
+				return
+			}
+
+			// dump logs from each containers
+			for _, container := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
```

Comment:
In the future we should limit the size of the logs we dump to something reasonable (say, 1MB) just in case there is some huge log in there.  (Ok to do in a later PR)